Sources/Events/EventsManager.swift

```diff
@@ -23,32 +27,25 @@ protocol EventsManagerType {
     func track(adEvent: AdEvent) async
     #endif
 
-    /// - Throws: if posting events fails
-    /// - Returns: the number of events posted
-    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
-    func flushEvents(batchSize: Int) async throws -> Int
-
-    /// - Throws: if posting feature events fails
-    /// - Returns: the number of feature events posted
     @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
-    func flushFeatureEvents(batchSize: Int) async throws -> Int
+    func flushEventsWithBackgroundTask() async
 
-    #if ENABLE_AD_EVENTS_TRACKING
-    /// - Throws: if posting ad events fails
-    /// - Returns: the number of ad events posted
     @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
-    func flushAdEvents(count: Int) async throws -> Int
-    #endif
+    func flushFeatureEventsWithBackgroundTask() async
```

Comment:
I don't have enough context on Ads around why this is a separate functoin from `flushEvents` and why there are instances where this is called instead of `flushEvents` but didn't want to break anything so I kept them separate